List structure

mmsesay opened the pull request:
In the PR the following was done:

- Setup webpack and css loader
- Setup linters configuration
- Add an array of objects with tasks
- Created a function to iterate through the tasks and render in the HTML element
- Invoke the function on page load

mmsesay committed:
Add: project structure
Add: linters configuration
Add: webpack configuration

mmsesay committed:
Add: style to project
Add: dymanic content from javascript
AddL stylint, webhint and eslint files

mmsesay committed:
Update readme.md with screenshot

mmsesay committed:
Update readme screenshot

mmsesay committed:
Fix: linter errors

mmsesay committed:
Fix: linter trailing comma error

rloterh commented on webpack.config.js:
```diff
@@ -0,0 +1,24 @@
+const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+module.exports = {
```
- [x] _Mode_ option is not set for _webpack_. Kindly set the `mode` and either assign its value to `development` or `production` within the _module.exports_

**Webpack config error:**
![image](https://user-images.githubusercontent.com/12745474/134145548-88d2d895-844b-466b-88a0-27c7b27a37a5.png)

mmsesay replied:
Thanks for picking this out. It has been resolved

rloterh commented on package.json:
```diff
@@ -0,0 +1,30 @@
+  "version": "1.0.0",
+  "description": "this is a simple todo list application using webpack and it is being served by webpack dev server",
+  "private": true,
+  "scripts": {
```
- [ ] A local server cannot be started within your application. Please add start options to your scripts to enable your application to be run using the command `npm start`, `npm run start` or `npm serve`.

**Package config error:**
![image](https://user-images.githubusercontent.com/12745474/134149164-5197482d-4a66-4d7d-ab06-d5c1a616529d.png)

mmsesay replied:
Thanks for picking this out. It has been resolved

rloterh commented on src/styles/style.css:
```diff
@@ -0,0 +1,58 @@
+body {
```
- [ ] OPTIONAL
Your application is not properly laid out in desktop versions. It appears at the bottom of the page. Kindly improve upon this aspect by ensuring that your app is accessible without scrolling the page.

**Your Application desktop view:**
![image](https://user-images.githubusercontent.com/12745474/134148559-1cfe4a77-4516-42bf-be0f-4a882f38c33b.png)

mmsesay replied:
Thanks for picking this out. It has been resolved

rloterh reviewed (changes requested):
Hi Muhammad, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍  
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly check the comments marked **[OPTIONAL]** under the review._
<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

mmsesay committed:
Add mode to development in webpack.config fle
Add: main script tag in the HTML
Add: style to reduce padding on desktop version

mmsesay committed:
Add: padding to the main html tag for desktop version

rloterh commented on webpack.config.js:
```diff
@@ -0,0 +1,25 @@
+module.exports = {
+  entry: './src/js/index.js',
+  mode: 'development',
+  plugins: [
```
- [ ] OPTIONAL
Kindly specify a `dev Sever` in your `Webpack.config.js` like this:

**Add a devServer to your Webpack:**
![image](https://user-images.githubusercontent.com/12745474/134181137-5d6d1c26-3623-4f21-9884-ce5150901041.png)

mmsesay replied:
Thanks. The dev server has been included.

rloterh commented on package.json:
```diff
@@ -0,0 +1,31 @@
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "webpack-dev-server --mode production --entry ./src/js/index.js --output-filename ./dist/bundle.js",
+    "build:prod": "webpack src/js/index.js dist/bundle.js -p",
+    "start": "webpack serve"
```
- [x] OPTIONAL
Your local server does open automatically when the command `npm start` is run. Kindly modify the start option to be:
`"start": "webpack serve --open",` 
to enable the page load automatically in the browser.

mmsesay replied:
I have added the --open to my start script

rloterh reviewed (approved):
## Status: Approved :heavy_check_mark:

Hi Muhammad, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾:shipit:.
Congratulations! 🎉🥂

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly consider my comment(s) marked **[OPTIONAL]** under this review:_

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
